rs/nns/governance/src/neuron_store.rs

```diff
@@ -1007,6 +1008,162 @@ impl NeuronStore {
         (active_neurons_in_stable_store, neuron_id_for_next_batch)
     }
 
+    /// Scans the contents of self, i.e. Neurons. This is basically range + map.
+    ///
+    /// I think this is good for use cases where you want to scan ALL neurons in
+    /// the background (i.e. not as part of servicing a request), but you do not
+    /// need to scan them all in one go. By "ALL" neurons, we mean regardless of
+    /// where they are stored (heap or stable memory).
+    ///
+    /// One strange limitation of this is that changes to neurons in stable
+    /// memory are not applied immediately. Instead, such changes are only made
+    /// after the scan. As long as fun does not look at neurons that it modified
+    /// in earlier calls, this weirdness is not observable, because by the time
+    /// this returns, all modifications will have been made.
+    ///
+    // Ideally, this would would not have map functionality, and instead just
+    // return an Iterator, but I'm not sure how to do that. This seems tough,
+    // because to get a StableNeuronStore, you have to call
+    // with_stable_neuron_store.
+    pub(crate) fn map_range(
+        &mut self,
+        neuron_ids: impl RangeBounds<NeuronId> + Copy,
+        neuron_sections: NeuronSections,
+
+        // This can make changes to Neurons as each one is scanned (via the
+        // "output" parameter).
+        //
+        // When this returns false, it means that scanning should stop.
+        // Therefore, to scan everything, arrange for this to always return
+        // true.
+        mut fun: impl FnMut(&mut Neuron) -> bool,
+    ) {
+        /// This does a couple of things for us:
+        ///
+        /// 1. Abstracts away the storage location of neurons.
+        ///
+        /// 2. In the case of stable memory, gives a way to remember what
+        ///    changes were made so they can be applied later. Whereas, changes to
+        ///    heap neurons can be made directly via &mut.
+        #[allow(clippy::large_enum_variant)]
+        #[derive(Debug)]
+        enum NeuronHandle<'a> {
```

Comment:
This is kind of a neat idea, but I wonder if we need to prune following for inactive neurons?  They're not getting voting rewards, and we plan to delete them soon (and not allow them to be resurrected).